airflow/providers/redis/sensors/redis_pub_sub.py

```diff
@@ -42,7 +42,7 @@ def __init__(self, *, channels: list[str] | str, redis_conn_id: str, **kwargs) -
         self.channels = channels
         self.redis_conn_id = redis_conn_id
         self.pubsub = RedisHook(redis_conn_id=self.redis_conn_id).get_conn().pubsub()
```

Comment:
IMO this should be moved to poke method too (maybe cached_property called from poke method)

Reply:
Would the intent just be to remove a method call from the init stage?   I could see it being a (cached?) property.

Reply:
> Would the intent just be to remove a method call from the init stage?

yes, currently the `RedisHook` is initiated on each time the DagFileProcessor parse the dag file.

Reply:
Yeah. Actually cached_property is best. Check it out.